easybuild/easyblocks/n/nvidia_compilers.py

```
@@ -0,0 +1,57 @@
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing NVIDIA HPC SDK compilers
+
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+from easybuild.easyblocks.generic.nvidiabase import NvidiaBase
+
+
+class EB_nvidia_minus_compilers(NvidiaBase):
+    """
+    Support for installing the NVIDIA HPC SDK (NVHPC) compilers
+    i.e. nvc, nvcc, nvfortran
+
+    Support for MPI or numeric libraries is disabled.
+    """
+
+    def prepare_step(self, *args, **kwargs):
+        """Prepare environment for installation."""
+        super().prepare_step(*args, **kwargs)
+
+        # Unsupported NVHPC options in nvidia-compilers are forced disabled
+        disabled_nvhpc_options = [
+            'module_byo_compilers',
+            'module_nvhpc_own_mpi',
+            'module_add_nccl',
+            'module_add_math_libs',
+        ]
+        for nvhpc_opt in disabled_nvhpc_options:
+            if self.cfg[nvhpc_opt]:
+                self.log.warning(f"Option '{nvhpc_opt}' forced to disabled in {self.name}-{self.version}")
```

I'd prefer that this immediately aborts the build instead of "silently" changing the option.
This only shows up in the logs, so people might not be aware of the changes (like I just did).

I'm also confused why we remove `NCCL` and the math libraries, but keep `NVSHMEM`?
I'd prefer to allow all three of them, but disable them by default.